include/oc_push.h

```diff
@@ -0,0 +1,232 @@
+/****************************************************************************
+ *
+ * Copyright 2021 ETRI All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * Created on: Aug 23, 2021,
+ * 				Author: Joo-Chul Kevin Lee (rune@etri.re.kr)
+ *
+ *
+ ****************************************************************************/
+
+#ifndef OC_PUSH_H
+#define OC_PUSH_H
+
+#include <stdio.h>
+
+#include "oc_config.h"
+#include "oc_helpers.h"
+#include "oc_rep.h"
+#include "oc_ri.h"
+#include "oc_endpoint.h"
+#include "port/oc_log.h"
+#include "util/oc_memb.h"
+#include "util/oc_process.h"
+
+
+#if defined(OC_PUSHDEBUG) || defined(OC_DEBUG)
+#ifndef OC_DEBUG
+#define oc_print(...) printf(__VA_ARGS__)
+#define oc_log(level, ...)                                                     \
+		do {                                                                         \
+			oc_print("%s: %s <%s:%d>: ", level, __FILE__, __func__, __LINE__);            \
+			oc_print(__VA_ARGS__);                                                        \
+			oc_print("\n");                                                               \
+		} while (0)
+#define p_dbg(...) oc_log("DEBUG", __VA_ARGS__)
```

Comment:
Lets keep the convention that macros are always all upper case. (I see that some log macros have small letter suffixes - such as `PRINTipaddr` - I'll fix that eventually. To keep other conventions it should be something like

p_dbg -> OC_PUSH_DBG
p_wrn -> OC_PUSH_WRN
p_err -> OC_PUSH_ERR